docs/preview/connected-registry/connected-registry-error-codes.md

```diff
@@ -0,0 +1,86 @@
+---
+title: Connected registry error code reference
+description: Details about error codes shown in the statusDetails property of a connected registry resource. For each error, possible solutions are listed.
+ms.topic: troubleshooting
+ms.date: 09/29/2021
+ms.author: jeburke
+author: jaysterp
+---
+
+# Connected registry error code reference
+
+This article helps you troubleshoot error codes you might encounter in the `StatusDetails` property of a connected registry.
+
+## Connection State
+
+The connection state of a connected registry indicates the current overall health status of the on-premises instance. The connection state is `Online` when the instance is healthy, `Offline` when the instance is not connected to internet, and `Unhealthy` when there is a critical error on the instance while it is online. When the connected registry resource has a connection state of `Unhealthy`, you may reference the `StatusDetails` property to view the corresponding error.
+
+## Status Details Format
+
+When your connected registry has a connection state of `Unhealthy` you can run the `az acr connected-registry show` command to view the statusDetails.
+
+`StatusDetails` provides a list of errors, each with the following format:
+
+```json
+{
+  "code": "Error code",
+  "correlationId": "CorrelationId of the error on the on-premises connected registry instance",
+  "description": "Description corresponding to this error",
+  "timestamp": "Timestamp corresponding to this error",
+  "type": "Component of the connected registry instance corresponding to the error"
+}
+```
+
+Every time the connected registry instance syncs with the cloud, these status details are updated. When the connected registry no longer has status details listed, it is considered healthy and its connection state is transitioned from `Unhealthy` to `Online`. Once the connected registry is no longer connected to internet, its connection state will transition to `Offline`.
+
+## Error Code Reference
+
+This section lists the possible codes you may see in the `StatusDetails` property of a connected registry, which indicate critical errors. For each error, possible solutions are listed.
+
+### DiskError
+
+This is the default error code when the connected registry is unable to create, write, or delete a file on the local disk. There are a few scenarios that may cause a `DiskError` code. Please reference below for possible scenarios and remediations. 
```

Comment:
Hust to make sure I understand: the `code` in the JSON above will have the value `DiskError` - is this correct? Shouldn't the code be numerical?

Reply:
Yes, the code is a string. We use a code in order to give a high level description of the error. For instance
```json
{
      "code": "DiskError",
      "correlationId": "73a46395-b89b-49c7-5621-d54e8b1574b5",
      "description": "Access to the path 'D:\\var\\acr\\data\\registry\\dummy.txt' is denied.",
      "timestamp": "2021-09-16T01:17:45.394512+00:00",
      "type": "Disk"
    }
```